### PR TITLE
Add WatershedLungspace class

### PR DESCRIPTION
**Pull request by psomhorst**
This PR adds the WatershedLungspace class. 

During development of this class some bugs were fixed and a lot of features were added, mainly to PixelMap and PixelMask. I also updated warnings calls in many files. I recommend going through this PR commit by commit, as I collected relevant changes in atomic commits. 

Documentation for the watershed method and a notebook demonstrating its use are included.

### CODE REVIEW

File: tests/test_watershed.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    assert np.nansum((tiv_mask - amplitude_mask).mask) == 0, "TIV mask should be a subset of amplitude mask"
+    assert np.nansum((tiv_mask - watershed_mask).mask) == 0, "TIV mask should be a subset of the watershed mask"
+    assert np.nansum((watershed_mask - amplitude_mask).mask) == 0, "TIV mask should be a subset of the watershed mask"
+
+    included_pixels = (~np.isnan(captures["mean tiv"].values)) & (~np.isnan(captures["mean amplitude"].values))
+    assert np.all(captures["mean tiv"].values <= captures["mean amplitude"].values, where=included_pixels)
+
+
+@pytest.mark.parametrize("threshold", [0.1, 0.15, 0.2, 0.25, 0.3])
+def test_watershed_with_simulated_data(simulated_eit_data: Callable[..., EITData], threshold: float):
```
Comment (JulietteFrancovich) [inline tests/test_watershed.py:108]: If peak_local_max returns an empty list (e.g., flat amplitude), the code may error. Maybe add a test with simulated data where amplitude is flat so no peaks are found, and assert the output mask is empty (all False) or raises a warning.

File: tests/test_watershed.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+
+@pytest.mark.parametrize("threshold", [-0.2, 0.0, 1.0, 1.2])
+def test_watershed_init_threshold_outside_range(threshold: float):
+    with pytest.raises(ValueError, match="Threshold must be between 0 and 1."):
+        _ = WatershedLungspace(threshold_fraction=threshold)
+
+
+@pytest.mark.parametrize("threshold", ["0.2", True, None])
+def test_watershed_init_thresolde_wrong_type(threshold: object):
```
Comment (JulietteFrancovich) [inline tests/test_watershed.py:78]:
```suggestion
def test_watershed_init_threshold_wrong_type(threshold: object):
```

File: tests/test_watershed.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+@pytest.mark.parametrize("threshold", [0.1, 0.15, 0.2, 0.25, 0.3])
+def test_watershed_with_real_data(draeger1: Sequence, threshold: float):
+    eit_data = draeger1.eit_data["raw"]
+    watershed_mask = WatershedLungspace(threshold_fraction=threshold).apply(eit_data, captures=(captures := {}))
+
+    assert captures, "captures should have values after runnning apply"
+    assert isinstance(watershed_mask, PixelMask)
+    assert np.nansum(watershed_mask.values) > 0
+    assert len(captures["local peaks"])  # There should be a non-empty list of local peaks
```
Comment (JulietteFrancovich) [inline tests/test_watershed.py:91]: Some captures are checked but not all are asserted. Is there a reason or should the creation and shape of all captures be checked?

File: eitprocessing/datahandling/pixelmap.py [hunk 4]
```diff
@@ -579,3 +579,24 @@ class SignedPendelluftMap(PixelMap):
     inflation starts), while negative values indicate pixels that have late inflation (after the global inflation
     starts).
     """
+
+
+@dataclass(frozen=True, init=False)
+class IntegerMap(PixelMap):
```
[the remark this answers is not in the document]
Comment (psomhorst) [reply eitprocessing/datahandling/pixelmap.py:617]: I added a visual test. 

Also, I adapted how an IntegerMap works, since the implementation was a bit hacky. Now, a PixelMap has a dtype argument, making it possible to change how values are casted on init.

Also, there was no unit test for integer maps.

File: eitprocessing/datahandling/pixelmap.py [hunk 5]
```diff
@@ -522,6 +522,38 @@ def _get_values(map_: npt.ArrayLike | PixelMap) -> np.ndarray:
         aggregated_values = aggregator(stacked, axis=0)
         return cls(values=aggregated_values, **return_attrs)
 
+    def to_boolean_array(self) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (psomhorst) [reply eitprocessing/datahandling/pixelmap.py:525]: Ik heb er `pm.to_boolean_array(zero=True)` van gemaakt. Evenzo `to_non_nan_array(nan=...)`, in lijn met `np.nan_to_num(nan=...)`.

File: eitprocessing/roi/watershed.py [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        # into a single region.
+        included_region = PixelMask(np.isin(watershed_regions, markers_inside_tiv_mask))
+        capture("included region", included_region)
+
+        if captures is not None:
+            capture("watershed regions", IntegerMap(watershed_regions))
+
+            included_marker_indices = np.isin(peaks_loc_int, markers_inside_tiv_mask)
+            included_peaks = np.argwhere(included_marker_indices)
+            excluded_peaks = np.argwhere(~included_marker_indices)
```
[the remark this answers is not in the document]
Comment (psomhorst) [reply eitprocessing/roi/watershed.py:160]: There are no background pixels in the watershed method. All pixels are associated with one of the peaks. I'm not sure what you are suggesting.

File: tests/test_watershed.py (hunk 1 above)
Comment (psomhorst) [reply tests/test_watershed.py:108]: If the amplitude is flat, breath detection would fail before this. I 'fixed' BreathDetection so it returns empty if no peaks are found at all. Then, WatershedLungspace will raise an exception.

File: eitprocessing/datahandling/eitdata.py [hunk 7]
```diff
@@ -144,6 +145,28 @@ def _sliced_copy(
     def __len__(self):
         return self.pixel_impedance.shape[0]
 
+    def get_summed_impedance(self, *, return_label: str | None = None, **return_kwargs) -> ContinuousData:
```
[the remark this answers is not in the document]
Comment (psomhorst) [reply eitprocessing/datahandling/eitdata.py:149]: The idea is that `get_summed_impedance` will replace the function of `calculate_global_impedance`. However, for now, `calculate_global_impedance` is used to calculate the global impedance of the raw signal. The idea is to mark `calculate_global_impedance` as deprecated in a future version, and remove it altogether in a later version.

Do you think we should already mark it deprecated in this PR?

File: eitprocessing/roi/tiv.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+from eitprocessing.datahandling.pixelmap import TIVMap
+from eitprocessing.features.pixel_breath import PixelBreath
+from eitprocessing.parameters.tidal_impedance_variation import TIV
+from eitprocessing.roi import PixelMask
+from eitprocessing.utils import make_capture
+
+
+@dataclass(kw_only=True, frozen=True)
+class TIVLungspace:
+    """Create a pixel mask by thresholding the mean TIV or amplitude.
```
Comment (JulietteFrancovich) [inline eitprocessing/roi/tiv.py:18]: Should the AmplitudeLungspace be a class on its own?

Comment (JulietteFrancovich) [reply eitprocessing/roi/tiv.py:18]: I'm not sure since the warning is very relevant that amplitude should probably not be used for FLS. However, if someone decides to anyway, it might be misleading if that results in a TIVLungspace object?

**Review by JulietteFrancovich — APPROVED** [review]
